fflib/src/classes/fflib_QueryFactory.cls

```diff
@@ -760,6 +790,18 @@ public class fflib_QueryFactory { //No explicit sharing declaration - inherit fr
 			this.setMessage( 'Invalid field \''+fieldName+'\' for object \''+objectType+'\'' );
 		}
 	}
+	
+	public class InvalidRelationshipException extends Exception
```

Comment:
I can't find anywhere else in this pull request where this exception is referenced. Perhaps it should be thrown if the passed in object type is not in the reference to list.

Reply:
Good one!!

Reply:
@squattingdog once you check this one we will merge it quickly.